### ISSUE TITLE
Create testing dataset of graphs of elementary functions

### ISSUE THREAD

**Issue by stasinos** [stasinos/machine_learning#4]
The testing data should be pairs of graph language inputs and the resulting graphs (bitmaps). As the purpose of the learner is to construct the definitions of the (hypothetically unknown) elementary functions sine(), sinc(), exp(), and ln(), the testing dataset only needs to have the graphs of the elementary functions in isolation.

One way to build the training dataset is to generate pairs of instantiated programs & their bitmaps by randomly filling the parameters that make sense for each function. Composition with poly() can be used for this.

**Comment by tatiana-boura** [comment]
I guess the second part is referring to the training set (?)

Also, the testing set will be something like that (?) :

1.  I :  sinc()    O :  bitmap of sinc(x) when $x \in [-9,8]$
2.  I :  sinc()    O :  bitmap of sinc(x) when $x \in [0,1]$
3.  I :  ln()    O :  bitmap  of ln(x) when $x \in [-1,1]$
4.  etc. ...

**Comment by stasinos** [comment]
> I guess the second part is referring to the training set (?)

No, it is a typo, it is still about the test set.

In order words, the testing set is like the training set (expresions , bitmaps), but for elementary functions only. The test is to see to what extent the system learned the expected elementary functions.

> Also, the testing set will be something like that (?) :

That was the original idea. However, the way the project is evolving this might need to be revised and start with (x,y) data, to avoid the complexity of generating and comparing bitmaps, and work with numerical (x,y) data. If this works really well and really quickly, we can revisit  bitmaps later.

**Comment by tatiana-boura** [comment]
Ok. So it looks like we are leaving this issue open for the time being.

**Comment by stasinos** [comment]
I am closing this issue, as the way the exercise progressed makes it redundant.
